GitFyle.Core.Api/Services/Foundations/Contributions/ContributionService.Validations.cs

```diff
@@ -0,0 +1,175 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright (c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using GitFyle.Core.Api.Models.Foundations.Contributions;
+using GitFyle.Core.Api.Models.Foundations.Contributions.Exceptions;
+
+namespace GitFyle.Core.Api.Services.Foundations.Contributions
+{
+    internal partial class ContributionService
+    {
+        private async ValueTask ValidateContributionOnAddAsync(Contribution contribution)
+        {
+            ValidateContributionIsNotNull(contribution);
+
+            Validate(
+                (Rule: await IsInvalidAsync(contribution.Id),
+                    Parameter: nameof(Contribution.Id)),
+
+                (Rule: await IsInvalidAsync(contribution.Title),
+                    Parameter: nameof(Contribution.Title)),
+
+                (Rule: await IsInvalidAsync(contribution.ExternalId), 
+                    Parameter: nameof(Contribution.ExternalId)),
+
+                (Rule: await IsInvalidAsync(contribution.RepositoryId),
+                     Parameter: nameof(Contribution.RepositoryId)),
+
+                (Rule: await IsInvalidAsync(contribution.CreatedBy), 
+                    Parameter: nameof(Contribution.CreatedBy)),
+
+                (Rule: await IsInvalidAsync(contribution.UpdatedBy), 
+                    Parameter: nameof(Contribution.UpdatedBy)),
+
+                (Rule: await IsInvalidAsync(contribution.ContributorId),
+                    Parameter: nameof(Contribution.ContributorId)),
+
+                (Rule: await IsInvalidLengthAsync(contribution.Title, 255), 
+                    Parameter: nameof(Contribution.Title)),
+
+                (Rule: await IsInvalidLengthAsync(contribution.ExternalId, 255), 
+                    Parameter: nameof(Contribution.ExternalId)),
+
+                (Rule: await IsInvalidAsync(contribution.ContributionTypeId), 
+                    Parameter: nameof(Contribution.ContributionTypeId)),
+
+                (Rule: await IsInvalidAsync(contribution.CreatedWhen), 
+                    Parameter: nameof(Contribution.CreatedWhen)),
+
+                (Rule: await IsInvalidAsync(contribution.UpdatedWhen), 
+                    Parameter: nameof(Contribution.UpdatedWhen)),
+
+                (Rule: await IsNotSameAsync(
+                    createBy: contribution.CreatedBy,
+                    updatedBy: contribution.UpdatedBy,
+                    createdByName: nameof(Contribution.CreatedBy)),
```

Comment:
Because the IsNotSameAsync method is generic we want to follow as suggested below.

```suggestion
                (Rule: await IsNotSameAsync(
                    first: contribution.CreatedBy,
                    second: contribution.UpdatedBy,
                    secondName: nameof(Contribution.CreatedBy)),
```